// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
package anezkasearch;

service IndexStorageService {
  rpc InsertIndex(KeyIndexes) returns (Empty);
  rpc GetIndexes(KeyIndexes) returns (KeyIndexes);
  rpc RemoveIndex(KeyIndexes) returns (Empty);
}

service SearchService {
  rpc SearchInd   ex(SearchRequest) returns (SearchResponse);
}

message KeyIndexes {
  string key = 1;

  repeated ToIndex indexes = 2;

  message ToIndex {
    oneof to{
      int64 i_ind = 3;
      string s_ind = 4;
    }
  }
}

message SearchRequest{
  string text = 1;
}

message SearchResponse{
  repeated ToIndex indexes = 1;

  message ToIndex {
    oneof to{
      int64 i_ind = 2;
      string s_ind = 3;
    }
  }
}


message Empty {}
